tests/optimagic/test_deprecations.py

```diff
@@ -862,6 +865,43 @@ def test_fun_with_dict_return_is_deprecated_in_slice_plot():
         )
 
 
+def test_handle_log_options():
+    msg = (
+        "Usage of the parameter log_options is deprecated "
+        "and will be removed in a future version. "
+        "Provide a Logger instance for the parameter `logging`, if you need to "
+        "configure the logging."
+    )
+    log_options = {"fast_logging": True}
+    with pytest.warns(FutureWarning, match=msg):
+        logger = None
+        handled_logger = handle_log_options_throw_deprecated_warning(
+            log_options, logger
+        )
+        assert handled_logger is None
+
+    creation_warning = (
+        f"\nUsing {log_options=} to create an instance of SQLiteLogOptions. "
+        f"This mechanism will be removed in the future."
+    )
+
+    with pytest.warns(match=creation_warning):
+        handled_logger = handle_log_options_throw_deprecated_warning(
+            log_options, ":memory:"
+        )
+        assert isinstance(handled_logger, SQLiteLogOptions)
+
+    incompatibility_msg = "Found string or path for logger argument, but parameter"
+    f" {log_options=} is not compatible "
+    log_options_typo = {"fast_lugging": False}
+
+    with pytest.raises(ValueError, match=incompatibility_msg):
+        handled_logger = handle_log_options_throw_deprecated_warning(
+            log_options_typo, ":memory:"
+        )
+        assert handled_logger == ":memory:"
```

Comment:
We are missing tests that make sure the old `log_options` still work but lead to a `FutureWarning` in:
- minimize
- maximize
- estimate_ml
- estimate_msm

Starting points for testing the estimation functions are [here](https://github.com/OpenSourceEconomics/optimagic/blob/a612912a7a186100fcf2bb1761f8d064ee0b4781/tests/optimagic/test_deprecations.py#L476)

Reply:
We also need such a test for using the old `LogReader`

Reply:
Done